src/server/graphql/models/authorization.js

```diff
@@ -42,6 +43,32 @@ export const requireSUOrSelf = (authToken, userId) => {
   throw errorObj({_error: 'Unauthorized. You cannot modify another user.'});
 };
 
+export const requireSUOrSelfOrLead = async (authToken, userId, teamId) => {
```

Comment:
i was thinking about it, but i'm still not sure what kinda workflow we're after for payment, so it could vary depending on that design sprint.